_episodes_rmd/01-r-basics.Rmd

```diff
@@ -0,0 +1,658 @@
+
+This is why programming languages often have a marker for missing data. `NA` is the special value in R that represents missing data. This marker lets the user know when missing data exists, but also provides ways to ignore the missing values in appropriate situations
+
+For example, if you try to do math on a numeric vector that has a missing value designated by `NA`, the function will return `NA` as the results to let you know that there is missing data.
+
+```{r missing-data}
+heights <- c(2, 4, 4, NA, 6)
+
+# Mean of a vector with a missing value?
+mean(heights)
+```
+
+This behavior is good because if you weren't expecting missing data, you want to know that it's there before doing more calculations. If you were expecting missing data and want to ignore if for the purposes of a calculation, many mathematical functions have the argument `na.rm`, which stands for remove missing data. By default, `na.rm = FALSE` which tells R not to ignore the missing data. Setting `na.rm = TRUE` will calculate the mean while ignoring the missing data. 
+
+```{r missing-data-narm}
+# Remove the missing data
+mean(heights, na.rm = TRUE)
+```
+
+However, not all functions have the `na.rm` argument. Luckily, R has other functions that identify and remove missing data.
+
+The `na.omit()` function takes an object as input returns the with the missing values removed.
+
+```{r na-omit}
+na.omit(heights)
+```
+Two other functions rely on logical vectors and the subset operator to remove missing values.
+
+* The `is.na()` function takes an object as input and  returns TRUE if the value its evaluating is NA. To remove `NA`s using `is.na()` you also need to use the not operator (`!`), which changes TRUE to false and vice versa.
+
+* The `complete.cases()` function takes an object as input and returns FALSE if the value is missing, so it can be used directly in the subset operator
+
+
+> ## Challenge: missing values
+>
+> Given the following vector
+>
+> ```r
+> ranks<-c(1,2,3,NA,"four")
+> ```
+> Which of the following statements would return a vector of length 3?
+>
+> ```1. na.omit(ranks)```
+> ```2. na.omit(as.numeric(ranks))```
+> ```3. ranks[complete.cases(ranks)]```
+> ```4. ranks[!is.na(ranks[ranks<4])]```
+>
+> > ## Solution
+> >
+> > 2
+> >
+> {: .solution}
+{: .challenge}
+
+# Glossary
+* R
+* RStudio
+* open source
+* cross platform
+* packages
+* reproducibility
+* script
+* console
+* environment
+* working directory
+* operators
+* arithmetic operator
+* assignment operator
+* functions 
+* arguments
+* data type
+* vector
+* element
+* coersion
+* subsetting
+* subset operator
+* slice operator
+* comparison operators
+* logical operators
+* missing values
```

Comment:
Maybe want to move the glossary in the extras of the lesson and/or link them to glossario. [Example in Jekyll lesson](https://carpentries-incubator.github.io/jekyll-pages-novice/reference.html)